syntax = "proto3";

package swift.auth;

import "common.proto";



// ============== 请求/响应 ==============

message RegisterRequest {
    string username = 1;
    string password = 2;
    string nickname = 3;
    string email = 4;
    string avatar_url = 5;
}

message RegisterResponse {
    int32 code = 1;
    string message = 2;
    string user_id = 3;
}

// 校验用户名密码，返回 user_id 与 profile（供 ZoneSvr 后接 OnlineSvr.Login）
message VerifyCredentialsRequest {
    string username = 1;
    string password = 2;
}

message VerifyCredentialsResponse {
    int32 code = 1;
    string message = 2;
    string user_id = 3;
    UserProfile profile = 4;
}

message GetProfileRequest {
    string user_id = 1;
}

message UpdateProfileRequest {
    string user_id = 1;
    string nickname = 2;
    string avatar_url = 3;
    string signature = 4;
}

// ============== 数据结构 ==============

message UserProfile {
    string user_id = 1;
    string username = 2;
    string nickname = 3;
    string avatar_url = 4;
    string signature = 5;
    int32 gender = 6;          // 0=未知, 1=男, 2=女
    int64 created_at = 7;
}

// ============== 服务定义 ==============

service AuthService {
    // 用户注册
    rpc Register(RegisterRequest) returns (RegisterResponse);

    // 校验用户名密码，返回 user_id 与 profile（登录时先调此接口，再调 OnlineSvr.Login）
    rpc VerifyCredentials(VerifyCredentialsRequest) returns (VerifyCredentialsResponse);

    // 获取用户资料
    rpc GetProfile(GetProfileRequest) returns (UserProfile);
    
    // 更新用户资料
    rpc UpdateProfile(UpdateProfileRequest) returns (swift.common.CommonResponse);
}
